EIPS/eip-5792.md

```diff
@@ -0,0 +1,257 @@
+wallets that support bundled transactions via [EIP-3074](./eip-3074.md). Dapps may use this more abstract interface to
+support different kinds of wallets, as well as provide a better UX for sending bundles of transactions (e.g. Approve and
+Call).
+
+## Motivation
+
+The current APIs to send transactions and check their status, `eth_sendTransaction` and `eth_getTransactionReceipt`,
+are not flexible enough for dapps to provide a good transaction user experience for sending transactions and showing
+their status. For example, these methods do not allow sending multiple transactions, or querying the status of a
+replaced-by-fee (sped up) transaction.
+
+This problem becomes worse with account abstracted wallets (e.g. [EIP-4337](./eip-4337.md)
+or [EIP-3074](./eip-3074.md)), where the actual transaction hash may not be known at the time of signing, but must still
+be returned by `eth_sendTransaction`. The transaction hash returned by `eth_sendTransaction` in these cases is unlikely
+to be relevant to the actual transaction hash of the included transaction. Dapps often link to the block explorer for
+the returned transaction hash, but in these cases the transaction hash is often wrong.
+
+Dapps need a better interface for sending batches of function calls from the user's wallet so the dapp can interact with
+wallets without considering the differences between wallet implementations. Some wallets may
+use [EIP-3074](./eip-3074.md) to deliver batches of messages as a single transaction, whereas others may use smart
+contract accounts via [EIP-4337](./eip-4337.md). Regular EOA wallets may send bundles of messages as individual
+transactions.
+
+The goal of the new interface is to be a more stable and flexible interface while wallet implementations evolve over
+time.
+
+## Specification
+
+Three new JSON RPC methods are added. Dapps may begin using these methods immediately, falling back
+to `eth_sendTransaction` and `eth_getTransactionReceipt` when they are not available.
+
+### `wallet_sendMessageBundle`
+
+Requests that the wallet deliver a group of messages on-chain from the user's wallet.
+The wallet MAY send all the messages as a single transaction.
+Dapps MUST NOT rely on the messages being sent in a single transaction, i.e. other untrusted transactions may be
+included between each of the transactions.
+The wallet MUST send these messages in the order specified in the request.
+The wallet MUST attempt to deliver all messages if it returns a successful response to this method, and the wallet MUST
+NOT deliver any messages if it returns an error response.
+The wallet MAY reject the request if the chain ID does not match the currently selected chain ID.
+The wallet MUST send the messages on the specified chain ID.
+The wallet MUST reject the request if the `from` address does not match the enabled account.
+The wallet MAY reject the request if one or more messages in the bundle will fail.
+
+#### Parameters
```

Comment:
Missing gas price parameters?

Reply:
Gas price is deliberately excluded. It's the wallet's responsibility to determine the gas price to get a bundle of messages in, especially because it can differ across multiple transactions and gas abstraction means the parameters may be ignored anyway. Wallets also already allow you to override gas prices, and it's a burden for dapps to set them.

Reply:
sgtm